packages/venia-concept/src/util/makeMediaUrl.js

```diff
@@ -0,0 +1,37 @@
+import memoize from 'lodash/memoize';
+
+const absoluteUrl = /^https?:\/\//i;
+
+const joinUrls = (base, url) =>
+    (base.endsWith('/') ? base.slice(0, -1) : base) +
+    '/' +
+    (url.startsWith('/') ? url.slice(1) : url);
```

Comment:
[Node's path module](https://nodejs.org/api/path.html#path_path_normalize_path) might be able to help here.